File: DeveloperTools/data2yaml/data/dmetro.lua
sptbl["dmetro"] = {

    files = { 
        module = "dmetro.c",
        header = "dmetro.h",
        example = "ex_dmetro.c",
    },
    
    func = {
        create = "sp_dmetro_create",
        destroy = "sp_dmetro_destroy",
        init = "sp_dmetro_init",
        compute = "sp_dmetro_compute",
    },
    
    params = {
        optional = {
            {
                name = "time",
                type = "SPFLOAT",
                description ="Time between triggers (in seconds). This will update at the start of each trigger.",
                default = 1.0
            },
        }
    },
    
    modtype = "module",
    
    description = [[Delta Metro. Produce a set of triggers spaced apart by  . 
time.]], 
    
    ninputs = 0,
    noutputs = 1,
    
    inputs = { 
        {
            name = "dummy",
            description = "this doesn't do anything."
        },
    },
    
    outputs = {
        {
            name = "out",
            description = "Trigger output."
        },
    }

}

File: AudioKit/Common/Internals/Soundpipe/modules/dmetro.c
#include <stdlib.h>
#include "soundpipe.h"

int sp_dmetro_create(sp_dmetro **p)
{
    *p = malloc(sizeof(sp_dmetro));
    return SP_OK;
}

int sp_dmetro_destroy(sp_dmetro **p)
{
    free(*p);
    return SP_OK;
}

int sp_dmetro_init(sp_data *sp, sp_dmetro *p)
{
    p->counter = 0;
    p->time = 1.0;
    return SP_OK;
}

int sp_dmetro_compute(sp_data *sp, sp_dmetro *p, SPFLOAT *in, SPFLOAT *out)
{
    *out = 0; 

    if(p->counter == 0) {
        *out = 1.0;
        p->counter = (int)(sp->sr * p->time) + 1;
    }

    p->counter--; 

    return SP_OK;
}
